### PR TITLE
LPS-117277 When the traffic is 0 the keywords link must be disabled

### PR DESCRIPTION
**Pull request by dgarciasarai**
**Description**

Disabled keywords link if traffic (organic or paid) is 0

**Steps to display Content Performance panel**

1. Connect your Liferay DXP with Analytics Cloud
2. Create a Display Page Template
3. Create a Web Content (and associate the previously created Display Page Template)
4. Create a Content Page and add and Asset Publisher
5. Click on the title of the Web Content to see the content
6. Click on the Content Performance icon in the control menu

**Feature flag: YES**

The Traffic Sources are hidden in the Content Performance panel. Configuration file:

File: `com.liferay.analytics.reports.web.internal.configuration.AnalyticsReportsConfiguration-default.config`

```
readsEnabled=B"true"
trafficSourcesEnabled=B"true"
```

**Module**

```
~/liferay-portal/modules/dxp/apps/analytics-reports/analytics-reports-web
```

**Screenshots**

Screenshot 2020-07-21 at 10 14 45

cc: @darquesdev

### CODE REVIEW

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TrafficSources.js [hunk 1]
```diff
@@ -125,9 +125,7 @@ export default function TrafficSources({
 												)
 											}
 										>
-											{!entry.countryKeywords.length ? (
-												<span>{entry.title}</span>
-											) : (
+											{entry.value > 0 ? (
```
Comment (cgoncas) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TrafficSources.js:128]: Could we have a test for it?

Comment (dgarciasarai) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TrafficSources.js:128]: Done 👍

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by darquesdev** [comment]
Great @dgarciasarai!

Now, despite the fact that a country is being sent within countryKeywords, navigation is disabled;)

```json
[
   {
      "countryKeywords":[
         {
            "keywords":[

            ],
            "countryCode":"us",
            "countryName":"United States"
         }
      ],
      "name":"paid",
      "share":0,
      "title":"Paid",
      "value":0,
      "helpMessage":"This number refers to the volume of people that find your page through paid keywords."
   },
   {
      "countryKeywords":[
         {
            "keywords":[

            ],
            "countryCode":"us",
            "countryName":"United States"
         }
      ],
      "name":"organic",
      "share":0,
      "title":"Organic",
      "value":0,
      "helpMessage":"This number refers to the volume of people that find your page through a search engine."
   }
]
```

image

**Comment by dgarciasarai** [comment]
ci:forward

**Comment by cgoncas** [comment]
Hey @dgarciasarai,

Until the pull is reviewed and Approved, we should never forward it.

We are not in a hurry, so it is better to wait until the pull has been reviewed.

As we have added new JS code to the JS component, should this pull be reviewed by someone in frontend as well?

Thanks!

**Pull request closed without merging**